Tests/FunctionalTests/util/FunctionalTestNetworkService.swift

```diff
@@ -89,37 +125,41 @@ class FunctionalTestNetworkService: NetworkService {
         return expectedNetworkRequests
     }
 
-    func setResponseConnectionFor(networkRequest: NetworkRequest, responseConnection: HttpConnection?) -> Bool {
-        for responseMatcher in responseMatchers {
-            if areNetworkRequestsEqual(lhs: responseMatcher.key, rhs: networkRequest) {
-                // unable to override response matcher
-                return false
-            }
+    /// Sets the mock `HttpConnection` response connection for a given `NetworkRequest`. Should only be used
+    /// when in mock mode.
+    ///
+    /// - Returns: `true` if the response was successfully set.
+    func setMockResponseFor(networkRequest: NetworkRequest, response: HttpConnection?) {
+        guard mockNetworkService else {
+            XCTFail("Setting mock responses when not in mock network service mode is not allowed.")
+            return
         }
-
-        // add new entry if not present already
-        responseMatchers[networkRequest] = responseConnection
-        return true
+        setResponseFor(networkRequest: networkRequest, responseConnection: response)
+    }
+    
+    // MARK: Network request response helpers
+    func getResponsesFor(networkRequest: NetworkRequest) -> [HttpConnection] {
```

Comment:
same comment for these helpers, we should clearly separate the two flows:
setMockResponseFor(... ) - sets mock response to be used with mockNetworkService
getReceivedResponseFor() - gets real responses when mockNetworkServices is off
**private** getMockResponseFor(...) - used only in this class for returning mock responses when mockNetworkService is on